consensus/state.go

```diff
@@ -97,11 +97,13 @@ type RoundState struct {
 }
 
 func (rs *RoundState) RoundStateEvent() types.EventDataRoundState {
+	// copy to avoid data race
```

Comment:
Why not have one routine with a select statement over the two pubsub channels and the quit channel ?